### PR TITLE
Introduce GitHub Actions for scylla-operator

### LINKED ISSUE #25: testing: add CI testing
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
Add CI testing before merging PRs.

**What is use case behind this feature:**
* Merge with confidence.
* Prevent regressions.

/cc @dahankzter

### PR DESCRIPTION
**Pull request by yanniszark**
**Description of your changes:**
Add the first CI workflow using GitHub actions!
This workflow:
- Checks formatting with fmt.
- Inspects code with vet.
- Ensures go dependencies are up-to-date.
- Runs unit tests.
- Builds the source code.

In addition, fix the version_upgrade unit tests.

**Which issue is resolved by this Pull Request:**
Resolves #25 

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Image has been built (`make docker-build`) on the last commit.

### CODE REVIEW

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Go
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (dahankzter) [inline .github/workflows/go.yaml]: What does "master" mean for pull requests? I mean the PRs are on it's own branch by definition.

File: .github/workflows/go.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        id: cache-tools
+        uses: actions/cache@v1
+        with:
+          path: ${{ format('{0}/bin/deps', env.working-directory-rel) }}
+          key: ${{ runner.os }}-${{ hashFiles(format('{0}/hack/binary_deps.py', env.working-directory-rel)) }}
+
+      - name: Install binary deps
+        if: steps['cache-tools'].outputs['cache-hit'] != 'true'
+        working-directory: ${{ env.working-directory }}
+        run: make bin/deps
```
Comment (dahankzter) [inline .github/workflows/go.yaml]: If this step takes too long it's quite possible for us to check in the binaries.

File: .github/workflows/go.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        working-directory: ${{ env.working-directory }}
+        run: hack/run_and_diff.py -v make test
+
+      - name: Build
+        working-directory: ${{ env.working-directory }}
+        run: make local-build
+
+# TODO:
+# - docker push?
+# - E2E
```
Comment (dahankzter) [inline .github/workflows/go.yaml]: Newline missing.

**Comment by dahankzter** [comment]
Good stuff! Let's put in that newline and merge!

File: .github/workflows/go.yaml (hunk 2 above)
Comment (yanniszark) [reply .github/workflows/go.yaml]: True, but ideally we don't want to go there.
Github Actions provide artifact caching and this seems like the ideal usecase (that's why I added it)

File: .github/workflows/go.yaml (hunk 1 above)
Comment (yanniszark) [reply .github/workflows/go.yaml]: According to this, I understand that the syntax means the workflow is triggered when a PR is made that targets the master branch.
https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags

File: .github/workflows/go.yaml (hunk 3 above)
Comment (yanniszark) [reply .github/workflows/go.yaml]: Fixed!

**Comment by yanniszark** [comment]
@dahankzter fixed!

**Pull request merged**